### ISSUE TITLE
Poor performance with good training/validation results

### ISSUE THREAD

**Issue by PhilReg** [ultralytics/yolov5#11546]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I want to train a rock, paper, scissor model. I have a very simple setup with just my webcam taking pictures infront of a white background with just my hands in the picture. I have 60 original images which I augmented up to 560. My training looks great, it seems like it should be performing very well but as soon as I load it up and try using it it barely recognizes anything with low confidence and extremly inaccurate. I guess it could be overfitting but the results are really terribly bad even for that, also I've tested it on some images from the trainingset and it detects some on those, but only with a confidence of about .50 which seems low from what the training indicates.

Example dataset images:
![2Paper d43cc58d-f4f3-11ed-937f-e8f40815b24b](https://github.com/ultralytics/yolov5/assets/116729488/fa5a5f82-1d7c-4aa6-8372-05f5bbda6ad7)
![1Scissor e9293943-f4f3-11ed-a94a-e8f40815b24b](https://github.com/ultralytics/yolov5/assets/116729488/5ce0dd46-3d07-4520-9f83-f81d01f29455)
![5Rock b0f8ba5e-f4f3-11ed-9299-e8f40815b24b](https://github.com/ultralytics/yolov5/assets/116729488/ef09a077-1363-4a8f-ac6c-2ba8f7d46b1e)

Training config: python train.py --img 320 --batch 16 --epochs 100--data dataset.yaml --weights yolov5s.pt --workers 8
Training history:
![results](https://github.com/ultralytics/yolov5/assets/116729488/eaffbabf-3703-49ff-a072-e96b2569e07f)
![confusion_matrix](https://github.com/ultralytics/yolov5/assets/116729488/3d3641fd-9e6a-44ae-ab13-0ac4f294346e)

Code to load model:
model = torch.hub.load('ultralytics/yolov5', 'custom', path='yolov5/runs/train/exp6/weights/best.pt')
No detections on test image:
![image](https://github.com/ultralytics/yolov5/assets/116729488/dc9732a2-c18a-4cb3-b0ed-65c1fb3471d6)

**Comment by glenn-jocher** [comment]
@PhilReg hello!

Have you tried adjusting the confidence threshold for detection in your model? The default value for YOLOv5 is 0.25 but depending on the use case, it might be necessary to adjust it to a higher value.

Also, how are you preprocessing your input images? It's important to ensure that they match the training data's preprocessing pipeline. If there's a mismatch in the data preparation, it could affect the performance of your model.

If these suggestions don't work, could you provide further details about your training process and a minimal reproducible example of the issue you're encountering? We would be happy to help you troubleshoot the performance issues you're experiencing.

Best regards,

**Comment by PhilReg** [comment]
Hi,
thank you very much for the input! It did end up being the confidence threshold. After adjusting it to a lower value the model is doing much better. 
![image](https://github.com/ultralytics/yolov5/assets/116729488/76b7a5f0-a647-447f-9daa-49584ef434ae)

**Comment by glenn-jocher** [comment]
@PhilReg hello!

Glad to hear that adjusting the confidence threshold helped improve the performance of your YOLOv5 model. Remember that tweaking hyperparameters like the confidence threshold can significantly impact the detection results, so it's important to experiment with different values and evaluate the performance of your model on a validation set.

If you have any further questions or issues, don't hesitate to reach out. We're here to help!

Best,
